samples/revoke_dataset_access.py

```diff
@@ -0,0 +1,43 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+def revoke_dataset_access(dataset_id, entity_id):
+
+    # [START bigquery_revoke_dataset_access]
+    from google.cloud import bigquery
+
+    # Construct a BigQuery client object.
+    client = bigquery.Client()
+
+    # TODO(developer): Set dataset_id to the ID of the dataset to fetch.
+    # dataset_id = 'your-project.your_dataset'
+
+    dataset = client.get_dataset(dataset_id)  # Make an API request.
+    entries = list(dataset.access_entries)
+
+    for entry in entries:
+        if entry.entity_id == entity_id:
+            entry.role = None
+            break
+
+    dataset.access_entries = entries
```

Comment:
Do we really have to replace the existing list with a new one with (at most) one of the elements modified, or could we just modify the element in-place? Wouldn't it work if we just directly iterated over `dataset.access_entries`?

Reply:
I'm uncertain if we support in-place edits. I think we don't at the moment because of our translation to/from API representations, but that'd be a good feature request.

Reply:
I checked the [setter code](https://github.com/googleapis/python-bigquery/blob/b8b5433898ec881f8da1303614780a660d94733a/google/cloud/bigquery/dataset.py#L442-L447) and it modifies internal `_properties`. On the other hand, [to_api_repr()](https://github.com/googleapis/python-bigquery/blob/b8b5433898ec881f8da1303614780a660d94733a/google/cloud/bigquery/dataset.py#L694-L700) merely returns a deep copy of these properties and does not try to compute the API representation, so this must happen when a value is assigned.

In other words, the new list indeed has to be re-assigned for the changes to take effect.